### PR TITLE
IGNITE-8176 Integrate gradient descent linear regression with partition based dataset

### PR DESCRIPTION
**Pull request by dmitrievanthony**

### CODE REVIEW

File: modules/ml/src/main/java/org/apache/ignite/ml/regressions/linear/LinearRegressionSGDTrainer.java [hunk 1]
```diff
@@ -17,51 +17,85 @@
...
-    }
+public class LinearRegressionSGDTrainer<P extends Serializable> implements SingleLabelDatasetTrainer<LinearRegressionModel> {
 
-    /** */
-    public LinearRegressionSGDTrainer(int maxIterations, double convergenceTol) {
-        this.gradientDescent = new GradientDescent(new LeastSquaresGradientFunction(), new BarzilaiBorweinUpdater())
-            .withMaxIterations(maxIterations)
-            .withConvergenceTol(convergenceTol);
-    }
+    private final UpdatesStrategy<? super MultilayerPerceptron, P> updatesStgy;
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/regressions/linear/LinearRegressionSGDTrainer.java]: missed javadoc in this file
